### DISCUSSION TITLE
No tools is given, but tries to use some tools that I haven't given

### DISCUSSION THREAD

**Discussion by ItzAmirreza** [crewAIInc/crewAI#1145] [Q&A]
Hey, I let the output speak, the title should explain the problem.
This only happens using gpt-3.5 turbo (latest version), but does NOT happen while doing the same with GPT-4. 
Note that the latest version of gpt-3.5 turbo has "improved instruction following" as they stated, so it really shouldn't be hallucinating tools.

```
Output
Entering new CrewAgentExecutor chain...
Thought: Do I need to use a tool? Yes
Action: Keyword Planner
Action Input: 'Find number of servers using plugin EzChestShop'Keyword Planner is not a valid tool, try one of [].Do I need to use a tool? Yes
Action: Google Trends
Action Input: 'Find number of servers using plugin EzChestShop'Google Trends is not a valid tool, try one of [].Do I need to use a tool? Yes
Action: Google Search
Action Input: 'Find number of servers using plugin EzChestShop'Google Search is not a valid tool, try one of [].Do I need to use a tool? No
Final Answer: 'Minecraft server plugin usage', 'EzChestShop server count', 'Number of servers with EzChestShop plugin', 'Server plugin usage statistics', 'EzChestShop plugin server count'
```

**Comment by ItzAmirreza** [comment]
Additional output (there you can see an error which is in blue in terminal 
```
 Entering new CrewAgentExecutor chain...
Thought: Do I need to use a tool? Yes
Action: Delegate work to co-worker
Action Input: Keyword Strategist|Generate relevant search terms for the query 'Find number of servers using plugin EzChestShop'
**Error executing tool. Missing exact 3 pipe (|) separated values. For example, `coworker|task|context`. I need to make sure to pass context as context.**
Do I need to use a tool? Yes
Action: Delegate work to co-worker
Action Input: Keywo
```

Error executing tool. Missing exact 3 pipe (|) separated values. For example, `coworker|task|context`. I need to make sure to pass context as context.

**Comment by quantumalchemy** [comment]
Yeah got this too:  Error executing tool. Missing exact 3 pipe (|) separated values. For example, `coworker|task|context`. I need to make sure to pass context as context. >> Did you fix it? How?   does it require some sort special of formatting in tasks ?

**Comment by ItzAmirreza** [comment]
Nope, the issue still exists, waiting for it to be addressed. Bump

**Comment by Haripritamreddy** [comment]
Can you send the code here?

**Comment by ItzAmirreza** [comment]
I just saw this commit which could be a fix for gpt3.5 turbo, not sure tho I will try it out and let you all know:
https://github.com/joaomdmoura/crewAI/commit/7f98a99e90f31792281d963ec9c94007aee9caa3

**Comment by nitishk94** [comment]
This issue still persists. I am using google gen ai to create an agent. It is using tools which are not passed or removed from the list of tools.
I passed the 3 tools FileReadTool, FileWriteTool, CodeInterpreterTool. I haven't seen in log of agent using FileWriterTool still my file gets overwritten. Then I removed the CodeInterpreterTool but It was showing in the output/log produced by agent that agent is accessing codeinterpretertool
